src/Microsoft.ApiDesignGuidelines.Analyzers/Core/UseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -0,0 +1,58 @@
+        internal const string RuleId = "CA1043";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersTitle), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersMessage), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Performance,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182180.aspx",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Property );
+
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var symbol = (IPropertySymbol)context.Symbol;
+            if ((symbol.IsIndexer) && !(symbol.IsOverride))
+            {
+                if (symbol.GetParameters().Length == 1)
+                {
+                    var paramType = symbol.GetParameters()[0].Type;
+                    if (!(paramType.SpecialType == SpecialType.System_String || paramType.SpecialType == SpecialType.System_Int32 || paramType.SpecialType == SpecialType.System_Int16 || paramType.SpecialType == SpecialType.System_Int64 || paramType.SpecialType == SpecialType.System_Object))
```

Comment:
this seems good candidate to extract out to its own local variable with good name to explain what this bool means.